src/extension/VSProject.cs

```diff
@@ -348,9 +348,13 @@ private void LoadMSBuildProject()
             MsBuildDocument = _doc;
 
             XmlElement assemblyNameElement = (XmlElement)_doc.SelectSingleNode("/msbuild:Project/msbuild:PropertyGroup/msbuild:AssemblyName", namespaceManager);
+            if (assemblyNameElement == null)
+                throw new ApplicationException("Missing required element AssemblyName");
             string assemblyName = assemblyNameElement.InnerText;
 
             XmlElement outputTypeElement = (XmlElement)_doc.SelectSingleNode("/msbuild:Project/msbuild:PropertyGroup/msbuild:OutputType", namespaceManager);
+            if (outputTypeElement == null)
+                throw new ApplicationException("Missing required element OutputType");
```

Comment:
We generally avoid direct use of ApplicationException. In the engine and extensions, we use NUnitEngineException, with an appropriate message.